### PR TITLE
Add event signin button feedback

### PR DESCRIPTION
**Pull request by audreygao**

### CODE REVIEW

**Comment by audreygao** [comment]
Screen Shot 2022-04-12 at 6 26 36 PM
Screen Shot 2022-04-12 at 6 26 50 PM

File: backend/src/controllers/EventController.ts [hunk 1]
```diff
@@ -230,6 +231,25 @@ export class EventController {
     return this.attendanceMapper.entityToResponse(newAttendance);
   }
 
+  @Get('/:eventID/signin/affiliate/result')
+  @ResponseSchema(AttendanceResultResponse)
+  @OpenAPI({ security: [{ TokenAuth: [] }] })
+  async affiliateEventSigninResult(
```
Comment (godwinpang) [inline backend/src/controllers/EventController.ts:237]: would be good to name functions as verbs, looks like this gets the attendance for an affiliate event?

```suggestion
  async getAffiliateEventAttendance
```

File: backend/src/payloads/Attendance.ts [hunk 2]
```diff
@@ -75,3 +75,14 @@ export class GetAttendanceQuery {
...
+
+export class AttendanceResultResponse {
+  @IsNumber()
+  attendeeId: number;
+
+  @IsNumber()
+  eventId: number;
+
+  @IsBoolean()
+  signin: boolean;
```
Comment (godwinpang) [inline backend/src/payloads/Attendance.ts:87]:
```suggestion
  isSignedIn: boolean;
```

File: frontend/src/pages/EventDetailsPage/components/EventDetails/index.tsx [hunk 3]
```diff
@@ -18,6 +18,9 @@ import { Tags, Card, GetLocation } from '@SharedComponents';
 import { EventResponse as EventInfo } from '@Services/api/models';
 import { EventStatusEnum } from '@Services/EventService';
 
+import { affiliateSignInToEventResult } from '@Services/EventService';
```
Comment (godwinpang) [inline frontend/src/pages/EventDetailsPage/components/EventDetails/index.tsx:21]: can consolidate with above import

File: frontend/src/pages/EventDetailsPage/components/EventDetails/index.tsx [hunk 4]
```diff
@@ -26,6 +29,15 @@ interface EventDetailsComponentProps {
 function EventDetailsComponent(props: EventDetailsComponentProps) {
   const { eventInfo, eventId } = props;
   const classes = useStyles();
+  const [result, setResult] = useState<boolean>(false);
```
Comment (godwinpang) [inline frontend/src/pages/EventDetailsPage/components/EventDetails/index.tsx:32]: nit: 
```suggestion
  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);

```

File: frontend/src/pages/EventDetailsPage/components/buttons/SignInButton.ts [hunk 5]
```diff
@@ -4,23 +4,42 @@ import { affiliateSignInToEvent } from '@Services/EventService';
...
+      onClick: () => {
+        try {
+          affiliateSignInToEvent(eventId);
+          alert("You've successfully signed in!");
+        } catch {
+          alert('Your sign in request could not be processed.');
+        }
+      },
+    });
+  }
```
Comment (godwinpang) [inline frontend/src/pages/EventDetailsPage/components/buttons/SignInButton.ts:40]:
```suggestion
  const buttonProps = signedIn ? {children: 'Signed In', disabled: true} : {children: 'Sign In'}
  return InducteeRenderPermission(Button)({
    ...buttonProps,
    primary: true,
    positive: true,
    onClick: () => {
      try {
        affiliateSignInToEvent(eventId);
        alert("You've successfully signed in!");
      } catch {
        alert('Your sign in request could not be processed.');
      }
    },
  });
```

**Review by godwinpang — CHANGES_REQUESTED** [review]

**Review by laurenkiyomi — DISMISSED** [review]

File: backend/src/payloads/Attendance.ts (hunk 2 above)
Comment (audreygao) [reply backend/src/payloads/Attendance.ts]: updated

File: frontend/src/pages/EventDetailsPage/components/buttons/SignInButton.ts [hunk 6]
```diff
@@ -23,4 +24,7 @@ function SignInButton({ eventId, children = 'Sign In' }: SignInButtonProps) {
   });
 }
 
+
+
+
```
Comment (godwinpang) [inline frontend/src/pages/EventDetailsPage/components/buttons/SignInButton.ts:29]: nit: it's good practice to eliminate the number of unnecessary changes to make code review easier
no need to change - just a small thing

**Review by godwinpang — APPROVED** [review]
thanks for the changes

**Pull request merged**